include/libopencm3/stm32/h7/memorymap.h

```diff
@@ -0,0 +1,158 @@
+#define QUADSPI_BASE        0x52005000U
+#define FMCC_BASE           0x52004000U
+#define JPEG_BASE           0x52003000U
+#define CHROMART_BASE       0x52001000U
+#define MDMA_BASE           0x52000000U
+#define GPV_BASE            0x51000000U
+#define FLASH_MEM_INTERFACE_BASE  0x52002000U
+
+/* APB3 Peripherals */
+#define WWDG1_BASE          0x50003000U
+#define LTDC_BASE           0x50001000U
+
+/* AHB2 Peripherals */
+#define DELAY_SDMMC2_BASE   0x48022800U
+#define SDMMC2_BASE         0x48022400U
+#define RNG_BASE            0x48021800U
+#define HASH_BASE           0x48021400U
+#define CRYPTO_BASE         0x48021000U
+#define DCMI_BASE           0x48020000U
+
+/* AHB1 Peripherals */
+#define USB2_OTG_FS_BASE    0x40080000U
+#define USB1_OTG_HS_BASE    0x40040000U
+#define ETHERNET_MAC_BASE   0x40028000U
+#define ADC1_ADC2_BASE      0x40022000U
+#define DMAMUX1_BASE        0x40020800U
+#define DMA2_BASE           0x40020400U
+#define DMA1_BASE           0x40020000U
+
+/* APB2 Peripherals */
+#define HRTIM_BASE          0x40017400U
+#define DFSDM1_BASE         0x40017000U
+#define SAI3_BASE           0x40016000U
+#define SAI2_BASE           0x40015C00U
+#define SAI1_BASE           0x40015800U
+#define SPI5_BASE           0x40015000U
+#define TIM17_BASE          0x40014800U
+#define TIM16_BASE          0x40014400U
+#define TIM15_BASE          0x40014000U
+#define SPI4_BASE           0x40013400U
+#define SPI1_BASE           0x40013000U
+#define USART6_BASE         0x40011400U
+#define USART1_BASE         0x40011000U
+#define TIM8_BASE           0x40010400U
+#define TIM1_BASE           0x40010000U
+
+/* APB1 Peripherals */
+#define CAN_MSG_BASE        0x4000AC00U
+#define CAN_CCU_BASE        0x4000A800U
+#define FDCAN2_BASE         0x4000A400U
+#define FDCAN1_BASE         0x4000A000U
+#define MDIOS_BASE          0x40009400U
+#define OPAMP_BASE          0x40009000U
+#define SWPMI_BASE          0x40008800U
+#define CRS_BASE            0x40008400U
+#define UART8_BASE          0x40007C00U
+#define UART7_BASE          0x40007800U
+#define DAC_BASE            0x40007400U
+#define HDMI_CEC_BASE       0x40006C00U
+#define I2C3_BASE           0x40005C00U
+#define I2C2_BASE           0x40005800U
+#define I2C1_BASE           0x40005400U
+#define UART5_BASE          0x40005000U
+#define UART4_BASE          0x40004C00U
+#define USART3_BASE         0x40004800U
+#define USART2_BASE         0x40004400U
+#define SPDIFRX1_BASE       0x40004000U
+#define SPI3_BASE           0x40003C00U
+#define SPI2_BASE           0x40003800U
+#define LPTIM1_BASE         0x40002400U
+#define TIM14_BASE          0x40002000U
+#define TIM13_BASE          0x40001C00U
+#define TIM12_BASE          0x40001800U
+#define TIM7_BASE           0x40001400U
+#define TIM6_BASE           0x40001000U
+#define TIM5_BASE           0x40000C00U
+#define TIM4_BASE           0x40000800U
+#define TIM3_BASE           0x40000400U
+#define TIM2_BASE           0x40000000U
```

Comment:
These have all _generally_ been definied in terms of APBx_BASE defines, but I don't really mind either way, as long as you've grouped them well enough as you have.